src/LiveComponent/doc/index.rst

```diff
@@ -2302,6 +2302,131 @@ You can also trigger a specific "action" instead of a normal re-render:
+
+You can bind your component LiveProp's values to URL query parameters thanks to the ``url`` option::
+
+    // src/Components/SearchModule.php
+    namespace App\Components;
+
+    use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+    use Symfony\UX\LiveComponent\Attribute\LiveProp;
+    use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+    #[AsLiveComponent]
+    class SearchModule
+    {
+        use DefaultActionTrait;
+
+        #[LiveProp(writable: true, url: true)]
+        public string $query = '';
+    }
+
+Now if you change the value of the ``query`` prop, the url will be updated to reflect the new state of your
+component, for example: ``https://my.domain/search?query=my+search+string``.
+
+Then, if you load this URL in your browser, your component will be initialized with the values provided in the query
+string.
+
+.. note::
+
+    URL is updated in place, no new entry is added to the browser's history.
+
+.. warning::
+
+    You can use multiple components with URL bindings in the same page, as long as bound field names don't collide.
+    Otherwise, you will observe unexpected behaviors.
+
+Supported data types
+~~~~~~~~~~~~~~~~~~~~
+
+You can use scalars, arrays and objects in your URL bindings:
+
+============================================  =================================================
+``prop`` value                                URL representation
+============================================  =================================================
+``'some search string'``                      ``prop=some+search+string``
+``42``                                        ``prop=42``
+``['foo', 'bar']``                            ``prop[0]=foo&prop[1]=bar``
+``{ foo: 'bar', baz: 42 }``                   ``prop[foo]=bar&prop[baz]=42``
+
+
+.. note::
+
+    Type consistency is enforced by the LiveComponent internal hydrator when the component is initialized with query
+    parameters. If a value could not be hydrated properly, it will be ignored to avoid unfriendly errors for the end
+    user.
+
+Multiple bindings
+~~~~~~~~~~~~~~~~~
+
+You can use as many URL bindings as you want in your component. To ensure the state is fully represented in the URL,
+all bound props will be set as query parameters, even if their values didn't change.
+
+For example, if you declare the following bindings::
+
+    // ...
+    #[AsLiveComponent]
+    class SearchModule
+    {
+        #[LiveProp(writable: true, url: true)]
+        public string $query = '';
+
+        #[LiveProp(writable: true, url: true)]
+        public string $mode = 'fulltext';
+
+        // ...
+    }
+
+
+And you only set the ``query`` value, then your URL will be updated to ``https://my.domain/search?query=my+query+string&mode=fulltext``.
+
+Validation
+~~~~~~~~~~
+
+.. caution::
+
+    It is recommended to validate data provided through query string parameters, so you can prevent malicious inputs
+    from being processed and rendered.
```

Comment:
This is interesting. If a `LiveProp` is writable, then the fact that it is bound to a query parameter does not open a new malicious entrypoint: the value can always be changed to "anything". Two things:

A) If we want to talk about validating writable properties in general more clearly, then we can talk about that in another PR
B) This section may still make sense, but I think it's only important for the (probably rare) situation where you have `url: true` and `writable: false` (e.g. when the user calls a `LiveAction`, you change the prop inside that method, but the user doesn't change it directly).

Reply:
The goal of this section was to be explicit about the fact that bound props are not validated when initialized from the URL, even if the component uses validation. But yes, the security concern only applies when the validation logic itself involves some security process (sanitizing or whatever). Maybe I can change the caution block to a note like:

`If your component uses validation, it is recommended to also validate data provided through query string parameters.`

For the B) part, I could add another example in this section to illustrate this use case, wdyt?

Reply:
Hmm, ok, yes, it's more clear to me. So the messaging would be something like:

> Like any writable `LiveProp`, because the user can modify this value, you should consider adding validation (<-- link to that section). When you bind a `LiveProp` to the URL, the initial value is *not* automatically validated. To validate it: (then the code).

Also, it's beyond the scope of the doc, but should the URL-binding feature validate automatically? And reject the value (skip it) if it's invalid?